### PR TITLE
Converts minor carp rift into a low requirement dynamic midround

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

There was an event which summoned a Carp Rift which ghosts could use to become carp.
As an event this was extremely unlikely to see use without an admin spawning it, just because the pool is so clogged. I have instead added it to the dynamic midround system.

## Mechanics

This is a _light_ antagonist which means it will attempt to spawn early or as a result of heavy rulesets failing.
It has almost no minimum requirements, it could occur in any round.
The midround event polls ghosts for candidates. If no ghosts want to be carp it doesn't run.
A random carp spawn point is selected and a Minor Carp Rift spawns there. These are outside of the station, but nearby (usually around the solar panels).
The ghost selected as the initial candidate spawns on top of it as a Mega Carp.

The Minor Carp Rift spawns an NPC carp every 30 seconds and can convert a ghost into a carp every 60 seconds (starting with one "in the tank"). Unused charges are pooled, so if a rift has been unused for 10 minutes the same ghost can spawn out of it 11 times (technically 10, the last charge can't be used by someone who has already taken one until a time limit has passed).
NPC carp spawned by this rift will start trying to path through the space station, and every carp from the same rift will have the same path in mind. Left alone, this means that they'll start causing problems for people in a specific department (perhaps more than one depending on where it sends them) until people are prompted to go do something about it.

Space Carp (as you likely know) can be taken out by two hits with any good makeshift weapon (a lit welder, or a bone saw).
The rift can be destroyed pretty simply by whacking it repeatedly with any weapon, though it takes a bit longer.

After 15 minutes of existence the carp rift changes colour and adds ghost carp charges twice as quickly, as well as spawning NPC carp more often. This counts as "winning" for the carp team.
Space Carp can talk, but are encouraged to play act as intelligent animals. What they do after "winning" is kind of up to them, but will probably end with the crew destroying the rift eventually.
While player Carp can try to be friendly (at their own risk) it's very difficult to tell them apart from the automatically-hostile NPC carp the rift will continue to spawn until it is destroyed.

## Concerns

This has a good chance of spawning too frequently, in which case we can take steps to reduce its priority or make it happen a maximum of once per round or something
It might actually just not be very fun
Unfortunately this is virtually impossible to test without just using a live environment

## Why It's Good For The Game

Adds something which isn't revenants or sleeper agents to the pool of things which might happen when there's low threat and a low player count.

## Changelog

:cl:
add: It's Carp Migration season, watch out for dangerous bluespace rifts to the Carp Stream opening in your area.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
After watching this happen in rounds I'm convinced this is actually _too_ unthreatening in its current form.
I am probably going to put this on hold until I merge some upstream changes to carp mobs which give them some more bite, and then do something like position the rift out in space while sending the automatically spawned carp on a path through the station.

**Comment by Jacquerel** [comment]
Carp rift now spawns outside, spawns an NPC carp every 30 seconds, and sets them on a migration path similar to the migration event.
Every spawned carp will try to take the same route, so will presumably annoy _someone_ into trying to do something about it.
Usually they will be in space (or on ice, in icebox) presenting a unique challenge for the crew.

**Review by lizardqueenlexi — APPROVED** [review]

**Pull request merged**
